swimster-api/s3.js

```diff
@@ -0,0 +1,57 @@
+const { S3Client, PutObjectCommand, GetObjectCommand } = require('@aws-sdk/client-s3')
+const { getSignedUrl } = require('@aws-sdk/s3-request-presigner')
+const crypto = require('crypto')
+const { getAwsS3Info } = require('./config')
+
+const s3BucketInfo = getAwsS3Info()
+const BYTES = 32
+
+const s3 = new S3Client({
+    region: s3BucketInfo.bucketRegion,
+    credentials: {
+        accessKeyId: s3BucketInfo.accessKey,
+        secretAccessKey: s3BucketInfo.secretKey
+    }
+})
+
+// creates a hash and attaches itself to image file name for unique identification
+const hashImageName = (bytes=BYTES, fileName) => crypto.randomBytes(bytes).toString('hex') + '_' + fileName
+
+// uploads an image to s3
+const uploadImageToS3 = async image => {
+    try {
+        const uploadParams = {
+            Bucket: s3BucketInfo.bucketName,
+            Key: hashImageName(BYTES, image.originalname),
+            Body: image.buffer,
+            ContentType: image.mimetype,
+        }
+        const result = await s3.send(new PutObjectCommand(uploadParams))
+        return {
+            result: result,
+            key: uploadParams.Key,
+        }
+    } catch (error) {
+        console.log("AWS Uploading Error:", error)
```

Comment:
intentional console log? usually these are just used for debugging and other messages. might be beter to do console.error()

Reply:
yes it was, will change now